orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,306 @@
+	{
+		// Set up DB adaptor using config.properties file
+		Object speciesToInferFromLong = "Homo sapiens";
+		String username = props.getProperty("username");
+		String password = props.getProperty("password");
+		String database = props.getProperty("database");
+		String host = props.getProperty("host");
+		int port = Integer.valueOf(props.getProperty("port"));
+		
+		dbAdaptor = new MySQLAdaptor(host, database, username, password, port);
+		// Statically store the adaptor variable in each class
+		InferReaction.setAdaptor(dbAdaptor);
+		SkipTests.setAdaptor(dbAdaptor);
+		GenerateInstance.setAdaptor(dbAdaptor);
+		OrthologousEntity.setAdaptor(dbAdaptor);
+		InferEWAS.setAdaptor(dbAdaptor);
+		UpdateHumanEvents.setAdaptor(dbAdaptor);
+		
+		SkipTests.getSkipList("normal_event_skip_list.txt");
+		
+		// For now an array of species names are used. I will likely change it so that the wrapper calls each organism individually during release.
+		ArrayList<String> speciesList = new ArrayList<String>(Arrays.asList("pfal", "spom", "scer", "ddis", "cele", "sscr", "btau", "cfam", "mmus", "rnor", "ggal", "tgut", "xtro", "drer", "dmel", "atha", "osat"));
+
+		JSONParser parser = new JSONParser();
+		Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+		JSONObject jsonObject = (JSONObject) obj;
+		for (String species : speciesList)
+		{
+			// Proper logging will be implemented before Release
+			JSONObject speciesObject = (JSONObject) jsonObject.get(species);
+			JSONArray speciesNames = (JSONArray) speciesObject.get("name");
+			String speciesName = (String) speciesNames.get(0);
+			System.out.println("Beginning orthoinference of " + speciesName + ".");
+			JSONObject refDb = (JSONObject) speciesObject.get("refdb");
+			String refDbUrl = (String) refDb.get("url");
+			String refDbProteinUrl = (String) refDb.get("access");
+			String refDbGeneUrl = (String) refDb.get("ensg_access");
+
+			JSONObject altRefDb = (JSONObject) speciesObject.get("alt_refdb");
+
+			// Creates two files that a) list reactions that are eligible for inference and b) those that are successfully inferred
+			String eligibleFilename = "eligible_" + species	+ "_75.txt";
+			String inferredFilename = "inferred_" + species + "_75.txt";
+			PrintWriter eligibleFile = new PrintWriter(eligibleFilename);
+			PrintWriter inferredFile = new PrintWriter(inferredFilename);
+			eligibleFile.close();
+			inferredFile.close();
+			InferReaction.setEligibleFilename(eligibleFilename);
+			InferReaction.setInferredFilename(inferredFilename);
+
+			// Set static variables (DB/Species Instances, mapping files) that will be repeatedly used
+			InferEvents.setInstanceEdits();
+			try {
+				HashMap<String,String[]> homologueMappings = InferEvents.readHomologueMappingFile(species, "hsap");
+				ProteinCount.setHomologueMappingFile(homologueMappings);
+				InferEWAS.setHomologueMappingFile(homologueMappings);
+				homologueMappings = new HashMap<String,String[]>();
```

Comment:
I was wondering why the homologueMappings variable is set to an empty HashMap here?

Reply:
Hmm that might be an artifact, or, more likely, it could have been an attempt at memory management (so still an artifact). In the summer there was a stretch of time where the memory usage of the script wasn't great, and I likely did this to null the mappings once they've been set. 

As for why I decided to do that by creating an empty HashMap instead of just using null, I do not know haha